### PR TITLE
 [FLINK-20096][docs] Clean up PyFlink docs

### PR DESCRIPTION
**Pull request by sjwiesman**
## What is the purpose of the change

This cleans up the PyFlink documentation. Mostly fixing English grammar and adding additional cross-links. I've also added an overview page to the section. 

Screen Shot 2020-11-11 at 1 22 22 PM

### CODE REVIEW

**Comment by sjwiesman** [comment]
cc @morsapaes

File: docs/dev/python/index.md [hunk 1]
```diff
@@ -22,3 +23,43 @@ KIND, either express or implied.  See the License for the
...
+* [Intro to PyFlink DataStream API]({% link dev/python/datastream_tutorial.md %})
+
+</div>
+<div class="col-sm-6" markdown="1">
+
+### Explore PyFlink
+
+The reference documentation covers all the details. Some starting points:
+
+* [PyFlink DataStream API]({% link dev/python/table-api-users-guide/index.md %})
```
Comment (dianfu) [inline docs/dev/python/index.md:57]: Nit: *PyFlink Table API* is located before the *Python DataStream API* in the *Try PyFlink* section. What about keeping the order consistent?

File: docs/dev/python/index.zh.md [hunk 2]
```diff
@@ -22,3 +23,43 @@ KIND, either express or implied.  See the License for the
...
+state and time, unlocking the full power of stream processing.
+
+<div class="row">
+<div class="col-sm-6" markdown="1">
+
+### Try PyFlink
+
+If you’re interested in playing around with Flink, try one of our tutorials:
+
+* [Intro to PyFlink Table API]({% link dev/python/table_api_tutorial.md %})
```
Comment (dianfu) [inline docs/dev/python/index.zh.md:47]:
```suggestion
* [Intro to PyFlink Table API]({% link dev/python/table_api_tutorial.zh.md %})
```

File: docs/dev/python/index.zh.md [hunk 3]
```diff
@@ -22,3 +23,43 @@ KIND, either express or implied.  See the License for the
...
+* [Intro to PyFlink DataStream API]({% link dev/python/datastream_tutorial.md %})
```
Comment (dianfu) [inline docs/dev/python/index.zh.md:48]:
```suggestion
* [Intro to PyFlink DataStream API]({% link dev/python/datastream_tutorial.zh.md %})
```

File: docs/dev/python/index.zh.md [hunk 4]
```diff
@@ -22,3 +23,43 @@ KIND, either express or implied.  See the License for the
...
+
+</div>
+<div class="col-sm-6" markdown="1">
+
+### Explore PyFlink
+
+The reference documentation covers all the details. Some starting points:
+
+* [PyFlink DataStream API]({% link dev/python/table-api-users-guide/index.md %})
```
Comment (dianfu) [inline docs/dev/python/index.zh.md:57]:
```suggestion
* [PyFlink DataStream API]({% link dev/python/table-api-users-guide/index.zh.md %})
```

File: docs/dev/python/index.zh.md [hunk 5]
```diff
@@ -22,3 +23,43 @@ KIND, either express or implied.  See the License for the
...
+* [PyFlink Table API &amp; SQL]({% link dev/python/datastream-api-users-guide/index.md %})
```
Comment (dianfu) [inline docs/dev/python/index.zh.md:58]:
```suggestion
* [PyFlink Table API &amp; SQL]({% link dev/python/datastream-api-users-guide/index.zh.md %})
```

**Review by dianfu — COMMENTED** [review]
@sjwiesman Thanks a lot for this great work! LGTM overall with just a few minor comments.

File: docs/dev/python/index.md [hunk 6]
```diff
@@ -22,3 +23,43 @@ KIND, either express or implied.  See the License for the
...
 under the License.
 -->
+
+<img src="{% link /fig/pyflink.svg %}" alt="PyFlink" class="offset" width="50%" />
+
+PyFlink is a language for building unified batch and streaming workloads.
+This means real-time streaming pipelines, performing exploratory data
+analysis at scale, building machine learning pipelines, and creating ETLs for a data platform.
+If you're already familiar with Python and libraries such as Pandas, then PyFlink makes it simple
+to leverage the full capabilities of the Apache Flink ecosystem.
```
Comment (morsapaes) [inline docs/dev/python/index.md:33]:
```suggestion
If you're already familiar with Python and libraries such as Pandas, then PyFlink makes it simpler to leverage the full capabilities of the Flink ecosystem. Depending on the level of abstraction you need, there are two different APIs that can be used in PyFlink:
```

File: docs/dev/python/index.md [hunk 7]
```diff
@@ -22,3 +23,43 @@ KIND, either express or implied.  See the License for the
...
+
+The PyFlink Table API makes it simple to write powerful relational queries for building reports and
+ETL pipelines.
+At the same time, the PyFlink DataStream API gives developers access to low-level control over
+state and time, unlocking the full power of stream processing.
```
Comment (morsapaes) [inline docs/dev/python/index.md:38]:
```suggestion
* The **PyFlink Table API** allows you to write powerful relational queries in a way that is similar to using SQL or working with tabular data in Python.

* At the same time, the **PyFlink DataStream API** gives you lower-level control over the core building blocks of Flink ([state](https://ci.apache.org/projects/flink/flink-docs-release-1.11/concepts/stateful-stream-processing.html#what-is-state) and [time](https://ci.apache.org/projects/flink/flink-docs-release-1.11/concepts/timely-stream-processing.html#introduction)) to build more complex stream processing use cases.
```

Comment (morsapaes) [reply docs/dev/python/index.md:38]: Think it's important here to give users something to identify with, like SQL and Pandas. Also, I don't think state and time in the way that we think about them are straightforward for Python users that don't know Flink.

**Review by morsapaes — COMMENTED** [review]
Thanks adding an overview page, Seth — this was much needed. Left some suggestions.

File: docs/dev/python/table-api-users-guide/index.md [hunk 8]
```diff
@@ -24,7 +24,6 @@ under the License.
 -->
 
 Python Table API allows users to develop [Table API]({% link dev/table/tableApi.md %}) programs using the Python language.
```
[the remark this answers is not in the document]
Comment (morsapaes) [reply docs/dev/python/table-api-users-guide/index.md:26]: In general, I'm confused whether the standard is to call it Python Table API or PyFlink Table API. As is, there's a mix of both in the docs.

File: docs/dev/python/index.md (hunk 7 above)
Comment (sjwiesman) [reply docs/dev/python/index.md:38]: I like that. For future reference, we don't cross-link via the actual url state should be: 

```md
[state]({% link concepts/stateful-stream-processing.md}#what-is-state)
```

**Comment by sjwiesman** [comment]
@morsapaes @dianfu thank you both for the review. I've tried to address all the comments, please take another look.

**Comment by sjwiesman** [comment]
Screen Shot 2020-11-12 at 10 24 01 AM

Updated overview for reference

**Comment by morsapaes** [comment]
Looks good from my side, thanks Seth!

**Comment by sjwiesman** [comment]
Grazi, Merging

**Pull request closed without merging**